clients/client/src/main/java/org/projectnessie/client/NessieClient.java

```diff
@@ -216,6 +217,16 @@ public Builder withPassword(String password) {
       return this;
     }
 
+    /**
+     * Set the token for {@link AuthType#BEARER} authentication.
+     *
+     * @param token bearer token
+     * @return {@code this}
+     */
+    public Builder withBearerToken(String token) {
```

Comment:
Is it the right approach to have specific methods for each auth type? It doesn't seem very extensible, and would force to handle more cases where this kind of method would be invoked with an invalid auth type. Maybe create a separate auth interface with various implementations to capture supported auth methods?

Reply:
Yes, I'm +1 to extracting individual interfaces for each auth type. This conflicts with #1816 though... which one do we want to merge first?

Reply:
#1819 is there as a draft (builds upon #1816). I don't mind merging this one first.